sdks/java/core/src/main/java/org/apache/beam/sdk/io/CountingInput.java

```diff
@@ -76,6 +86,16 @@ public static BoundedCountingInput upTo(long numElements) {
   }
 
   /**
+   * Creates a {@link BoundedCountingInput} that will produce the specified number of elements,
+   * starting from {@code startOffset} to (excluding) {@code endOffset}.
+   */
+  public static BoundedCountingInput forSubrange(long startOffset, long endOffset) {
```

Comment:
I think "index" might be a better name than "offset" here - offsets are usually measured relative to something important (e.g. the beginning of a file).